event-sourcing/event-repository/event-repository-jdbc/src/test/java/uk/gov/justice/services/eventsourcing/repository/jdbc/eventstream/EventStreamJdbcRepositoryIT.java

```diff
@@ -0,0 +1,260 @@
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+
+import uk.gov.justice.services.eventsourcing.repository.jdbc.exception.InvalidSequenceIdException;
+import uk.gov.justice.services.jdbc.persistence.JdbcRepositoryException;
+import uk.gov.justice.services.test.utils.persistence.AbstractJdbcRepositoryIT;
+
+import java.sql.SQLException;
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Stream;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class EventStreamJdbcRepositoryIT extends AbstractJdbcRepositoryIT<EventStreamJdbcRepository> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(EventStreamJdbcRepository.class);
+
+    private static final String LIQUIBASE_EVENT_STORE_DB_CHANGELOG_XML = "liquibase/event-store-db-changelog.xml";
+
+    public EventStreamJdbcRepositoryIT() {
+        super(LIQUIBASE_EVENT_STORE_DB_CHANGELOG_XML);
+    }
+
+    @Before
+    public void initializeDependencies() throws Exception {
+        jdbcRepository = new EventStreamJdbcRepository();
+        jdbcRepository.logger = LOGGER;
+        registerDataSource();
+    }
+
+    @Test
+    public void shouldStoreEventStreamUsingInsert() throws InvalidSequenceIdException {
+        jdbcRepository.insert(eventStreamOf(randomUUID(), 1L));
+        jdbcRepository.insert(eventStreamOf(randomUUID(), 2L));
+        jdbcRepository.insert(eventStreamOf(randomUUID(), 3L));
+
+        final Stream<EventStream> streamOfStreams = jdbcRepository.findAll();
+
+        assertThat(streamOfStreams.count(), equalTo(3L));
+    }
+
+
+    @Test(expected = JdbcRepositoryException.class)
+    public void shouldThrowExceptionOnDuplicateStreamId() throws InvalidSequenceIdException {
+        final UUID streamID = randomUUID();
+        jdbcRepository.insert(eventStreamOf(streamID, 1L));
+        jdbcRepository.insert(eventStreamOf(streamID, 2L));
+    }
+
+    @Test
+    public void shouldReturnPageOfSpecifiedSize() throws InvalidSequenceIdException, SQLException {
+
+        final UUID streamId1 = randomUUID();
+        final UUID streamId2 = randomUUID();
+        final UUID streamId3 = randomUUID();
+        final UUID streamId4 = randomUUID();
+        final UUID streamId5 = randomUUID();
+        final UUID streamId6 = randomUUID();
+        final UUID streamId7 = randomUUID();
+
+        jdbcRepository.insert(eventStreamOf(streamId1, 1L));
+        jdbcRepository.insert(eventStreamOf(streamId2, 2L));
+        jdbcRepository.insert(eventStreamOf(streamId3, 3L));
+        jdbcRepository.insert(eventStreamOf(streamId4, 4L));
+        jdbcRepository.insert(eventStreamOf(streamId5, 5L));
+        jdbcRepository.insert(eventStreamOf(streamId6, 6L));
+        jdbcRepository.insert(eventStreamOf(streamId7, 7L));
+
+        final List<EventStream> streams = jdbcRepository.previous(4L, 3L).collect(toList());
```

Comment:
Can we maybe pull the 4L and 3L into local variables, so it's clear that we are requesting streamId/pagesize.

Also maybe pageSize should be an int. Would also help to clarify.

Reply:
Fixed